### ISSUE TITLE
Delete key does not work in direct overlay

### ISSUE THREAD

**Issue by phosphene47** [ppy/osu#6944]
**Describe the bug:** 

Delete key (not backspace) does not work in the direct overlay's search textbox. However, it works with `Ctrl` modifier pressed, and deletes the entire word as expected.

I believe this is a bug with `SearchableListFilterControl` because beatmap selection search produces the same behavior.

**Screenshots or videos showing encountered issue:** 

https://streamable.com/clmiz

Delete key was pressed 10 times in this video.

**osu!lazer version:** 2019.1122.0

**Logs:** 
[runtime.log](https://github.com/ppy/osu/files/3883640/runtime.log)

**Comment by bdach** [comment]
Looks like it's explicitly suppressed here:

https://github.com/ppy/osu/blob/f43e4bfd703d4cd3a9f747a7ead88ac4e690f08c/osu.Game/Graphics/UserInterface/SearchTextBox.cs#L37-L41

([see framework-side `GlobalAction` mapping](https://github.com/ppy/osu-framework/blob/7f7dd04779bfa45d33d69a1c5599c894a9c64d33/osu.Framework/Platform/GameHost.cs#L888))

**Comment by Histmy** [comment]
I noticed that the delete key still doesn't do anything in the beatmap listing. After some digging, I came across this issue. I was wondering if a fix for this is planned?

I went through the whole UI (I may have missed some areas) and found that this behavior is the same in many places, but also in many places the key just works like expected. I compiled this list:

Delete key behaves as expected:
 - Language and skin search in settings
 - Chat overlay (F8)
 - Comment section in changelog overlay
 - Collection search (under singleplayer beatmap search)
 - Room name and password in multiplayer room creation
 - Chat in multiplayer room
 - Chat in playlist room
 - Chat in daily challenge
 - Probably everywhere in the beatmap editor

Delete key behaves like @luaneko described:
 - Search bar in settings
 - Search in beatmap listing overlay
 - Both search boxes in the dashboard (F9)
 - Music search overlay (F6)
 - Singleplayer beatmap search
 - Search in Manage collections
 - Search for multiplayer lounge
 - Search for playlist lounge

I wasn't able to figure out this project's layout, but I did read the comment linked by @bdach in the code (still present in a similar form today). If I understand correctly, it states that the delete key doesn't need to be handled because the caret cannot move backwards using arrow keys. However, this only applies to the search bar in settings and the singleplayer beatmap search. In all other cases, text inputs allow caret movement using arrow keys. And also moveing the caret with the mouse cursor works with every one of them. So I don't see why the delete key shouldn't work in all of them.

TL;DR: The behavior of the text boxes in the game is inconsistent. Some allow caret movement with arrow keys, while others don't. Also, some allow forward deleting with the delete key, and others don't. Would it be possible to standardize this please?
